packages/@react-aria/test-utils/src/tree.ts

```diff
@@ -0,0 +1,248 @@
+        }
+
+        // Note that long press interactions with rows is strictly touch only for grid rows
+        await triggerLongPress({element: cell, advanceTimer: this._advanceTimer, pointerOpts: {pointerType: 'touch'}});
+      } else {
+        await pressElement(this.user, cell, interactionType);
+      }
+    }
+  };
+
+  /**
+   * Toggles the expansion for the specified tree row. Defaults to using the interaction type set on the tree tester.
+   */
+  async toggleRowExpansion(opts: TreeToggleExpansionOpts) {
+    let {
+      row,
+      interactionType = this._interactionType
+    } = opts;
+    if (!this.tree.contains(document.activeElement)) {
+      await act(async () => {
+        this.tree.focus();
+      });
+    }
+
+    if (typeof row === 'string' || typeof row === 'number') {
+      row = this.findRow({rowIndexOrText: row});
+    }
+
+    if (!row) {
+      throw new Error('Target row not found in the tree.');
+    } else if (row.getAttribute('aria-expanded') == null) {
+      throw new Error('Target row is not expandable.');
+    }
+
+    if (interactionType === 'mouse' || interactionType === 'touch') {
+      let rowExpander = within(row).getAllByRole('button')[0]; // what happens if the button is not first? how can we differentiate?
+      await pressElement(this.user, rowExpander, interactionType);
+    } else if (interactionType === 'keyboard') {
+      if (row?.getAttribute('aria-disabled') === 'true') {
+        return;
+      }
+
+      await this.keyboardNavigateToRow({row});
+      if (row.getAttribute('aria-expanded') === 'true') {
+        await this.user.keyboard('[ArrowLeft]');
+      } else {
+        await this.user.keyboard('[ArrowRight]');
+      }
+    }
+  };
+
+  /**
+   * Triggers the action for the specified tree row. Defaults to using the interaction type set on the tree tester.
+   */
+  async triggerRowAction(opts: TreeRowActionOpts) {
+    let {
+      row,
+      needsDoubleClick,
+      interactionType = this._interactionType
+    } = opts;
+
+    if (typeof row === 'string' || typeof row === 'number') {
+      row = this.findRow({rowIndexOrText: row});
+    }
+
+    if (!row) {
+      throw new Error('Target row not found in the tree.');
+    }
+
+    if (needsDoubleClick) {
+      await this.user.dblClick(row);
+    } else if (interactionType === 'keyboard') {
+      if (row?.getAttribute('aria-disabled') === 'true') {
+        return;
+      }
+
+      if (document.activeElement !== this._tree || !this._tree.contains(document.activeElement)) {
+        act(() => this._tree.focus());
+      }
+
+      await this.keyboardNavigateToRow({row});
+      await this.user.keyboard('[Enter]');
+    } else {
+      await pressElement(this.user, row, interactionType);
+    }
+  };
+
+  /**
+   * Returns the tree.
+   */
+  get tree():  HTMLElement {
+    return this._tree;
+  }
+
+  /**
+   * Returns the tree's rows if any.
+   */
+  get rows(): HTMLElement[] {
```

Comment:
Something to consider API-wise: some of the test utils provide getters that return rows/items/options, and some of them are methods (e.g. select/menu/listbox) I guess because they support the optional filter parameter. We could consider making these match for consistency. Also I assume we might eventually support sections in table/tree/etc. as well.

Reply:
yeah, this is something I've flip flopped on several times across several iterations of the test utils :/. I kinda wanna gauge how useful people even think the `filter` parameter is. If it turns out not to be very useful, I'm leaning towards making them all getters

Reply:
I've mostly used `rows.filter(...)` in order to filter
is there some case that can't cover? or is it just convenience?